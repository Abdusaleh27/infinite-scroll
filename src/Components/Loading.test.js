import Loading from "./Loading";
import { render, screen } from "@testing-library/react";
import React from "react";
//smoke test
it("Loading component renders", () => {
  render(<Loading />);
});

//snapshot test
it("Loading component snapshot test", () => {
  const { asFragment } = render(<Loading />);
  expect(asFragment()).toMatchSnapshot();
});

//contains the loading indicators tests

test("Loading component contains loading indicators", () => {
  render(<Loading />);
  expect(screen.getByRole("status")).toBeInTheDocument();
  expect(screen.getByText("Loading...")).toBeInTheDocument();
});
