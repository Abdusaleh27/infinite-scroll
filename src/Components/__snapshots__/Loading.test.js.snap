// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component snapshot test 1`] = `
<DocumentFragment>
  <div
    class="spinner-border"
    role="status"
  >
    <span
      class="visually-hidden"
    >
      Loading...
    </span>
  </div>
</DocumentFragment>
`;
